Tweak repo sidebar

silverwind opened the pull request:
Before and after:

<img width="218" alt="Screenshot 2024-12-15 at 04 53 53" src="https://github.com/user-attachments/assets/299b1f0a-ba72-47c6-b662-a9d540d4d741" />
<img width="222" alt="Screenshot 2024-12-15 at 04 53 41" src="https://github.com/user-attachments/assets/5a2b5332-e324-4d20-82e9-21d1c850e826" />

Diff without whitespace: https://github.com/go-gitea/gitea/pull/32847/files?diff=unified&w=1

The `tw-mt-2` is fine even if the element renders empty:

<img width="387" alt="image" src="https://github.com/user-attachments/assets/76a976e4-ba2e-48a5-9248-c361552a937a" />

wxiaoguang commented on templates/repo/home_sidebar_top.tmpl:
```diff
@@ -41,28 +41,30 @@
-					</a>
-				</div>
-			{{end}}
+			<div class="tw-flex tw-flex-col tw-gap-1 tw-mt-2">
```
I do not think it is right to use other manually written "flex" elements in the "flex-list", it breaks the framework design and will cause more regressions in the future.

silverwind replied:
What do you suggest instead to increase gap between those links?

wxiaoguang replied:
Drop "flex-list" completely if it doesn't fit this use case.

silverwind replied:
Could be done I guess but it will be a more involved refactor.

wxiaoguang replied:
Yes , I also agree with "separate div", but the "flex-body" shouldn't be wrapped by another manually written "flex" element. At least, our "devtest page" doesn't say it is well supported.

Gitea has many legacy UI problems, just because the UI was heavily patched by breaking a lot of framework design (including breaking Fomantic UI).

wxiaoguang replied:
> I'm against putting these links straight into the top-level div. Putting them in a separate div will be much better.

"flex-list" means `<div class="flex-list">`, by framework design

wxiaoguang commented:
By reading the tmpl code again, I can see the "flex-list" was already abused.

Ideally it should be changed to a layout like this:

```
<div class="flex-list">
    <flex-item>
         <flex-item-main>
              <flex-item-title> Description (the section title)</>
              <flex-item-body with other styles> 
                    description
                    topic
                    any other elements etc ....
              </>
          </>
     </>
</>

```

silverwind commented:
Did some tweaks:

<img width="402" alt="image" src="https://github.com/user-attachments/assets/77807781-a3b1-440c-a8bb-c10094fd8c2d" />

- Link wrapper div is now `flex-item-body`
- Individual links are `flex-text-block`
- Tweaked homepage link so it's colored like GitHub's.

wxiaoguang commented:
Thank you, will correct the "flex-list" based on your work

silverwind commented:
One tiny tweak added, gap between topic labels is reduced. I'll let you finish now.

<img width="290" alt="image" src="https://github.com/user-attachments/assets/d0ae9d1a-7820-4b44-af5b-dab7e2ce0f40" />

wxiaoguang commented:
Force pushed (because too many changes) and applied your "gap-1" for the labels.

silverwind commented:
Bug: label text is too dull because of `color: inherit`:

<img width="301" alt="image" src="https://github.com/user-attachments/assets/93c46c40-102d-4414-bcd3-d9dbe72fd85d" />

silverwind commented:
Fix pushed: 

<img width="299" alt="image" src="https://github.com/user-attachments/assets/ca754a7b-a4c0-454b-8ee9-f761836ec7a8" />

silverwind commented on templates/repo/home_sidebar_top.tmpl:
```diff
@@ -8,61 +8,61 @@
-					<button id="manage_topic" class="btn interact-fg tw-text-12">{{ctx.Locale.Tr "repo.topic.manage_topics"}}</button>
-				{{end}}
-			{{end}}
-			{{if and .Permission.IsAdmin (not .Repository.IsArchived)}}
```
This condition removal is fine? The diff is hard to read.

wxiaoguang replied:
It has been merged to the block beside the "manage topic" button. They are related.

![image](https://github.com/user-attachments/assets/befc88b3-b2cd-4110-ae4f-4cda99378bb0)

wxiaoguang commented:
Now it should be clearer than before

* every element is in a "flex-item-body"
* no need to patch the "flex-list" styles

<details>

![image](https://github.com/user-attachments/assets/ad10b48a-37e4-4ab3-a54e-6fba994b8a46)

</details>

silverwind commented:
One final tweak done to link and its hover effect:

<img width="204" alt="image" src="https://github.com/user-attachments/assets/4bad3c43-3b78-4866-b42b-2e2d538c974f" />

silverwind commented:
Did more tweaks:

<img width="312" alt="image" src="https://github.com/user-attachments/assets/663fcab7-e84c-432d-b8e0-1eebd947d825" />


- Restored 16px font size for description block, it fits better with latest release which also has 16px from `.flex-item-title`.
- Extracted release label into its own file. This was previously always showing untranslated "latest" while the release itself would show "Stable" on the release page. Now it shows "Latest" on both pages and the translation for that is added.
- Moved the `repo-home-sidebar-bottom` and `repo-home-sidebar-top` divs into their respective templates.

wxiaoguang commented:
> * Extracted release label into its own subtemplate. This was previously always showing untranslated "latest" (i18n key was missing) while the release itself would show "Stable" on the release page. Now it shows "Latest" on both pages and the translation for that is added.

That's not right. There could be many "stable" but only one "latest"

![image](https://github.com/user-attachments/assets/00ae92d1-8f68-4417-8e0d-6f91cd26476b)

silverwind commented:
Hmm so we need to somehow adjust the label template to decect latest vs. stable.

wxiaoguang commented:
> Hmm so we need to somehow adjust the label template to detect latest vs. stable.

I would suggest do not do that trick in this PR.

Leave the "latest" on repo home, and leave the "stable" for the list.

silverwind commented:
OK:

Repo:
<img width="141" alt="image" src="https://github.com/user-attachments/assets/43db3dcb-38ff-49d3-9ca6-ea68988b2a17" />

Release:
<img width="126" alt="image" src="https://github.com/user-attachments/assets/e3be968d-86ff-41d6-bc65-488112ba002c" />

silverwind commented:
Ideally this `IsLatest` attribute would be set in backend, but I have no idea where to change that.

wxiaoguang commented:
> Ideally this `IsLatest` attribute would be set as `.Release.IsLatest` in backend, but I have no idea where to change that.

Since the time for releasing 1.23 is limited, I would suggest we only do necessary changes at the moment, keep 1.23 release stable. Leave trivial (or hard) problem to the future.
